openjdk/barriers/mmtkObjectBarrier.cpp

```diff
@@ -2,6 +2,21 @@
 #include "mmtkObjectBarrier.hpp"
 #include "runtime/interfaceSupport.inline.hpp"
 
+void MMTkObjectBarrierSetRuntime::object_probable_write(oop new_obj) const {
+#if MMTK_ENABLE_BARRIER_FASTPATH
+  intptr_t addr = (intptr_t) (void*) new_obj;
+  uint8_t* meta_addr = (uint8_t*) (SIDE_METADATA_BASE_ADDRESS + (addr >> 6));
+  intptr_t shift = (addr >> 3) & 0b111;
+  uint8_t byte_val = *meta_addr;
+  if (((byte_val >> shift) & 1) == 1) {
+    // only promoted objects will reach here
+    mmtk_object_probable_write((MMTk_Mutator) &Thread::current()->third_party_heap_mutator, (void*) new_obj);
```

Comment:
The method includes the fast path, and it will check the log bit again. Performance-wise it may not be a concern, but it can cause confusion. This also is different from the following normal write barrier code which calls different methods. If this is intended, please add some comments to explain the reason behind the decision.

Reply:
Updated comments to explain the fast-path unlog bit check.